controller/sync.go

```diff
@@ -168,6 +177,16 @@ func (m *appStateManager) SyncAppState(app *v1alpha1.Application, state *v1alpha
 		}),
 		sync.WithSyncWaveHook(delayBetweenSyncWaves),
 		sync.WithPruneLast(syncOp.SyncOptions.HasOption("PruneLast=true")),
+		sync.WithResourceModificationChecker(syncOp.SyncOptions.HasOption("ApplyOutOfSync=true"), func(target *unstructured.Unstructured, live *unstructured.Unstructured) bool {
+			diffResult, err := diff.Diff(target, live,
```

Comment:
Please see suggestion in https://github.com/argoproj/gitops-engine/pull/213#pullrequestreview-578825696


The `diffResults` is available in `CompareAppState` method and can be returned in an additional field of `comparisonResult` struct:
https://github.com/argoproj/argo-cd/blob/3967baf0c95fa2678f499f23a773e19249265d99/controller/state.go#L82